### PR TITLE
Add Spotless to Gradle

### LINKED ISSUE #4817: Add Spotless to Gradle
### Problem

We need to reach feature parity with the Maven build before we can replace it with Gradle.

### Solution

* Add Spotless plugin to Gradle
* Add license header updating to spotless
* Add version replacement functionality

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by steven-sheehy**
**Description**:

* Add git gradle plugin for git metadata in actuator endpoints
* Add `gradle.properties` version update to Maven
* Add `release` task to Gradle for version replacment
* Add Spotless plugin to auto-format `.java`, `.js`, `.json`, `.kts`, `.md`, `.sh`, `.sql`, `.xml`, `.yaml`, `.yml`
* Add license header automatically to files
* Add a Git hook so Spotless gets run automatically on commit and auto-formats staged files
* Auto download node & npm for use by Prettier that's invoked by Spotless for some file types
* Fix cleanup of target directory necessary for the Gradle Docker build to work
* Fix tagging pre-release tags as `latest` via Gradle
* Remove download of license and source jars from Maven
* Remove husky, pretty-quick, and prettier

**Related issue(s)**:

Fixes #4817

**Notes for reviewer**:

After approval and merge, will comment out ratchetFrom and format all files in repo. Then we can remove `StripOldLicenseFormatterStep.kt`.

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

**Review by MarcKriguerAtHedera — APPROVED** [review]
LGTM.

File: gradle.properties [hunk 1]
```diff
@@ -2,7 +2,7 @@
 dockerRegistry=gcr.io/mirrornode
 dockerTag=latest
 group="com.hedera"
-version=0.68.0-SNAPSHOT
+version=0.70.0-SNAPSHOT
```
Comment (xin-hedera) [inline gradle.properties:5]:
```suggestion
version=0.71.0-SNAPSHOT
```

**Review by xin-hedera — APPROVED** [review]
LGTM. just the incorrect snapshot version

File: gradle.properties (hunk 1 above)
Comment (steven-sheehy) [reply gradle.properties:5]: Will address in a follow up Gradle related PR. Or it will resolve itself on next release.

**Pull request merged**